core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java

```diff
@@ -74,7 +74,7 @@ private <T extends Identity> SdkHttpFullRequest sraSignRequest(SdkHttpFullReques
 
         ExecutionAttributes executionAttributes = context.executionAttributes();
         SelectedAuthScheme<T> selectedAuthScheme =
-            executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
+            (SelectedAuthScheme<T>) executionAttributes.getAttribute(SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME);
```

Comment:
There are 2 casts here that I need help to find alternatives too.

Reply:
If you haven't come up with anything new since our offline discussion, I think the casts are ok.